src/main/java/com/iota/iri/controllers/TransactionViewModel.java

```diff
@@ -309,28 +355,30 @@ public TransactionViewModel getTrunkTransaction(Tangle tangle) throws Exception
 
     /**
      * @return The trits stored in the {@link TransactionViewModel} . If the trits aren't available in the controller,
-     * the trits are generated from the transaction object and stored in the controller {@link #trits()} variable.
+     *         the trits are generated from the transaction object and stored in the controller {@link #trits()}
+     *         variable.
      */
     public synchronized byte[] trits() {
         return (trits == null) ? (trits = trits(transaction.bytes)) : trits;
     }
 
     /**
      * Deletes the {@link Transaction} object from the database
-     * 
+     *
      * @param tangle The tangle reference for the database
      * @throws Exception Thrown if there is an error removing the object
      */
     public void delete(Tangle tangle) throws Exception {
         tangle.delete(Transaction.class, hash);
+        cacheEvict(tangle, hash);
```

Comment:
I am trying to think whether there is a threading problem...
I would add the first step is to make sure fresh and dirty are false
So that there is no way that after you delete from tangle that this will be persisted again upon eviction

Reply:
Deleting from tangle also delete from the cache. So the tvm will not be available during eviction to DB